[New] Add dynamic entity layer (#127)
Esri/arcgis-maps-sdk-swift-samples: Shared/Samples/Add dynamic entity layer/README.md

# Add dynamic entity layer

Display data from an ArcGIS stream service using a dynamic entity layer.

| AddDynamicEntityLayer | Settings |
| ----------------- | ------------- |
|![Image of add dynamic entity layer](AddDynamicEntityLayer.png)|![Image of add dynamic entity layer settings](AddDynamicEntityLayer_Settings.png)|

## Use case

> A stream service is a type of service provided by ArcGIS Velocity and GeoEvent Server that allows clients to receive a stream of data observations via a web socket. ArcGIS Runtime allows you to connect to a stream service and manage the information as dynamic entities within the application, and display them in a dynamic entity layer. Displaying information from feeds such as a stream service is important in applications like dashboards where users need to visualize and track updates of real-world objects in real-time.

Reviewer: ```suggestion
A stream service is a type of service provided by ArcGIS Velocity and GeoEvent Server that allows clients to receive a stream of data observations via a web socket. ArcGIS Maps SDK allows you to connect to a stream service and manage the information as dynamic entities and display them in a dynamic entity layer. Displaying information from feeds such as a stream service is important in applications like dashboards where users need to visualize and track updates of real-world objects in real-time.
```

`common-samples/pull/3829`

Use `ArcGISStreamService` to manage the connection to the stream service and purge options to manage how much data is stored and maintained by the application. The dynamic entity layer will display the latest received observation, and you can set track display properties to determine how to display historical information for each dynamic entity. This includes the number of previous observations to show, whether to display track lines in-between previous observations, and setting renderers.

## How to use the sample

Use the controls to connect to or disconnect from the stream service, modify display properties in the dynamic entity layer, and purge all observations from the application.

## How it works

> 1. Create an `ArcGIStreamService` using a `Uri`.

Reviewer: ```suggestion
1. Create an `ArcGIStreamService` using a stream service URL.
```
Most of the time if a design is not created by a Swift team member, it could contain phrasing that is specific to other platforms (in this case, .NET). Make sure to read through the whole README and make adjustments.

p.s., When migrate from an existing example, compare the old iOS README and the design - sometimes there are minor discrepancies between them. Pick the content that makes the most sense.

> 2. Set a `DynamicEntityFilter` on the stream service to limit the amount of data coming from the server.

Reviewer: I don't see the `DynamicEntityFilter` type in the API. Can you double check (with .NET, presumably Kerry) if this is the correct step, or we should use `ArcGISStreamServiceFilter` instead?

Author: Yup should be `ArcGISStreamServiceFilter`.

Reviewer: Then it can be
```suggestion
2. Set an `ArcGISStreamServiceFilter` on the stream service to limit the amount of data coming from the server.
```

> 3. Set the `MaximumDuration` property of the stream service `PurgeOptions` to limit the amount of data managed by the application.

Reviewer: ```suggestion
3. Set the `maximumDuration` property of the stream service's `purgeOptions` to limit the amount of data managed by the application.
```

4. Create a `DynamicEntityLayer` using the stream service.

> 5. Update values in the layer's `TrackDisplayProperties` to customize the layer's appearance.

Reviewer: ```suggestion
5. Update values in the layer's `trackDisplayProperties` to customize the layer's appearance.
```

6. Add the `DynamicEntityLayer` to the map.

## Relevant API

* ArcGISStreamService
* ConnectionStatus
* DynamicEntity

> * DynamicEntityFilter

Reviewer: As above, please check if this type is used.

* DynamicEntityLayer

> * DynamicEntityPurgeOptions

Reviewer: It seems this type is nested in Swift API, comparing to .NET. Change it to the correct type and reorder the list both here and in metadata.
```
* DynamicEntityDataSource.PurgeOptions
```

* TrackDisplayProperties

## About the data

This sample uses a [stream service](https://realtimegis2016.esri.com:6443/arcgis/rest/services/SandyVehicles/StreamServer) that simulates live data coming from snowplows near Sandy, Utah. There are multiple vehicle types and multiple agencies operating the snowplows.

## Additional information

> More information about dynamic entities can be found in the [guide documentation](link goes here).

Reviewer: Please discuss with realtime questions folks to get the link that should go here.

Reviewer: Thanks. Let's remove the Additional information section from README. Once it is updated in the design I can add it back here.
